### ISSUE TITLE
Show results from Chromium master

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt.fyi#406]
https://ci.chromium.org/p/chromium has a number of builders which run the imported web-platform-tests. Getting these results into wpt.fyi would be valuable for at least these reasons:
* Time from Chromium CL landing to results is *much* shorter than waiting for Chrome Dev or Chrome Canary.
* Chromium has virtual test suites with different flags, and visualizing those results is currently hard.
* Identifying unknown differences between `run_web_tests.py` (content_shell) and `./wpt run`  (chrome).

The virtual test suites point led @Orphis to create https://adam.orphis.net/~orphis/wpt.html for the WebRTC team to check how the test suite would fare with transceivers (a new WebRTC primitive) enabled.

**Comment by foolip** [comment]
This would require:
 * `run_web_tests.sh` able to output a wptreport.json, or one created after the fact.
 * That artifact being saved for long enough for it to be sent to wpt.fyi (@Hexcles is working on similar integration for Taskcluster)
 * Supporting the notion of virtual test suites, perhaps simply be mapping them into a directory like Chromium.

**Comment by foolip** [comment]
I discussed this more with @Hexcles. @Hexcles, can you summarize roughly what we'd have to do to extract results from the Chromium tree?

**Comment by foolip** [comment]
An entirely different approach would be to run *Chromium* builds from https://ci.chromium.org/p/chromium/builders/luci.chromium.ci/linux-rel. gs://chromium-browser-snapshots/Linux_x64/584988/chrome-linux.zip is an example. I can't find a chromedriver binary though.

**Comment by Hexcles** [comment]
Sorry for the late reply. Philip mentioned a "different" approach above. The original approach that we discussed was to extract test results from TestExpectations and baselines at ToT.

That'd require some sophisticated parsing to recover subtest information from baselines of testharness tests (as Blink test runner simply treats the whole output as a text file for comparison). And even in the best scenario, we still can't know the full list of subtests for a test file if all of them pass as we don't check in all-PASS baselines. A workaround is to special-case it on the frontend (e.g. the result says "all-PASS" and the frontend fills PASS into all cells of that column).

**Comment by lukebjerring** [comment]
Bumping this old thread. Is this still something we're looking at?
@foolip I think there was an endpoint available for getting a chromedriver binary given a Chrome version?

**Comment by foolip** [comment]
If we could get this to work I think it would be very valuable. However, I think we should consider it blocked on @LukeZielinski's work on using wptrunner in Chromium CI, as that would make this fairly trivial.

**Comment by foolip** [comment]
@lukebjerring yes there is some way of getting a chromdriver binary given a Chrome version. Quoting what @johnchen0 has told me over email:
> - First, find out which version of Chrome you are using. Let's say you have Chrome 70.0.3538.22.
> - Take the Chrome version number, remove the last part, and append the result to URL "https://chromedriver.storage.googleapis.com/LATEST_RELEASE_". For example, with Chrome version 70.0.3538.22, you'd get a URL "https://chromedriver.storage.googleapis.com/LATEST_RELEASE_70.0.3538".
> - Use the URL created in the last step to retrieve a small file containing the version of ChromeDriver to use. For example, the above URL will get your a file containing "70.0.3538.16". (The actual number may change in the future, of course.)
> - Use the version number retrieved from the previous step to construct the URL to download ChromeDriver. With version 70.0.3538.16, the URL would be https://chromedriver.storage.googleapis.com/index.html?path=70.0.3538.16/.
> - After the initial download, it is recommended that you occasionally go through the above process again to see if there are any bug fix releases.

@johnchen0, is this documented somewhere in public?

**Comment by LukeZielinski** [comment]
Good to know that WPT will need to continue to generate wptreport.json from Chromium CI. wpt.fyi expects a single file for the entire run, right? If so, we'll need to make sure to merge the results across shards to create a single file. Do we have code that does this already?

**Comment by Hexcles** [comment]
@LukeZielinski no, I fixed the results receiver so it can handle (merge) multiple reports correctly now.

**Comment by JohnChen0** [comment]
> @JohnChen0, is this documented somewhere in public?

See https://groups.google.com/forum/#!topic/chromedriver-users/HvlKWqv34es. This versioning model is not yet official, so no better documentation than the announcement. I expect to make it official within a month, with the ChromeDriver web site updated for it.

**Comment by foolip** [comment]
I see, thanks!

**Comment by Hexcles** [comment]
This has been fixed by https://github.com/web-platform-tests/wpt/pull/24904
